library/src/main/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionedRecyclerViewAdapter.java

```diff
@@ -793,6 +793,106 @@ void callSuperNotifyItemMoved(int fromPosition, int toPosition) {
         super.notifyItemMoved(fromPosition, toPosition);
     }
 
+    /**
+     * Helper method that calls {@link #notifyItemInserted} when header changed to visible.
+     *
+     * @param tag unique identifier of the section
+     */
+    public void notifyHeaderInsertedInSection(String tag) {
+        int headerPosition = getHeaderPositionInAdapter(tag);
+
+        callSuperNotifyItemInserted(headerPosition);
+    }
+
+    /**
+     * Helper method that calls {@link #notifyItemInserted} when header changed to visible.
+     *
+     * @param section a visible section of this adapter
+     */
+    public void notifyHeaderInsertedInSection(Section section) {
+        int headerPosition = getHeaderPositionInAdapter(section);
+
+        callSuperNotifyItemInserted(headerPosition);
+    }
+
+    /**
+     * Helper method that calls {@link #notifyItemInserted} when footer changed to visible.
+     *
+     * @param tag unique identifier of the section
+     */
+    public void notifyFooterInsertedInSection(String tag) {
+        int footerPosition = getFooterPositionInAdapter(tag);
+
+        callSuperNotifyItemInserted(footerPosition);
+    }
+
+    /**
+     * Helper method that calls {@link #notifyItemInserted} when footer changed to visible.
+     *
+     * @param section a visible section of this adapter
+     */
+    public void notifyFooterInsertedInSection(Section section) {
+        int footerPosition = getFooterPositionInAdapter(section);
+
+        callSuperNotifyItemInserted(footerPosition);
+    }
+
+    /**
+     * Helper method that calls {@link #notifyItemInserted} when section changed to visible.
+     *
+     * @param tag unique identifier of the section
+     */
+    public void notifyVisibilityChangedToVisible(String tag) {
+        Section section = getSection(tag);
+
+        notifyVisibilityChangedToVisible(section);
+    }
+
+    /**
+     * Helper method that calls {@link #notifyItemRangeRemoved} when section changed to invisible.
+     *
+     * @param tag unique identifier of the section
+     * @param previousSectionPosition previous section position using {@link #getSectionPosition}
```

Comment:
I would remove the "using {@link #getSectionPosition}" from here because is not clear that the user should call it before setting the position to false. Move this explanation to the description of the method 👍